src/+groundwater/calculateGroundwaterRecharge.m

```diff
@@ -0,0 +1,131 @@
+
+                                %%%%%%%%%% Variables definitions %%%%%%%%%%
+        Equations of the water balance are implemented in this function (Equations 8-13 of HYDRUS-MODFLOW paper) as follows:
+        recharge = recharge_init + (SY - sy) * DeltZ, where:
+
+        gwfluxes                structure that includes the outputs (groundwater recharge and its individual components)
+        recharge                groundwater recharge, which is the upper boundary flux of the top layer of the phreatic aquifer (after the correction of the specific yield)
+        recharge_init           upper boundary flux into the moving balancing domain (before the correction of the specific yield)
+        SS                      specific storage of MODFLOW aquifers
+        SY                      large-scale specific yield of the phreatic aquifer
+        sy                      small-scale specific yield (dynamically changing water yield) caused by fluctuation of the water table
+        depToGWT_strt           soil layer thickness from the top of the soil up to the phreatic surface at the start of the current time step
+        indxGWLay_strt          index of the soil layer that includes the phreatic surface at the start of the current time step
+        depToGWT_end            soil layer thickness from the top of the soil up to the phreatic surface at the end of the current time step
+        indxGWLay_end           index of the soil layer that includes the phreatic surface at the end of the current time step
+        Theta_L                 soil moisture at the start of the current time step (bottom to top)
+        Theta_LL                soil moisture at the end of the current time step (bottom to top)
+        STheta_L                soil moisture at the start of the current time step (top to bottom)
+        STheta_LL               soil moisture at the end of the current time step (top to bottom)
+        soilThick               cumulative soil layers thickness (from top to bottom)
+        indxAqLay               index of the MODFLOW aquifer that corresponds to each STEMMUS soil layer
+        aqLayers                elevation of top surface level and all bottom levels of aquifer layers, received from MODFLOW through BMI
+    %}
+
+    % Load model settings
+    ModelSettings = io.getModelSettings();
+    NN = ModelSettings.NN; % Number of nodes
+    NL = ModelSettings.NL; % Number of layers
+
+    % Start Recharge calculations
+    if GroundwaterSettings.GroundwaterCoupling == 1 % Groundwater coupling is enabled
+        % Load needed variables
+        soilThick = GroundwaterSettings.soilThick; % cumulative soil layer thickness (from top to bottom)
+        indxAqLay = GroundwaterSettings.indxAqLay; % index of MODFLOW aquifer layers for each STEMMUS soil layer
+        aqLayers = GroundwaterSettings.aqLayers; % elevation of top surface level and all bottom levels of aquifer layers
```

Comment:
Please move these line where they are used in the script.

Reply:
> Please move these lines where they are used in the script.
>
Okay, done